### PR TITLE
[New Rule] Process Termination Followed by Deletion

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/481

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by Samirbous** [comment]
image

File: rules/windows/defense_evasion_process_termination_followed_by_deletion.toml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+maturity = "production"
+updated_date = "2020/11/04"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies process termination followed by the deletion of its executable file. Malware, tools, or other non-native
+files dropped or created on a system by an adversary may leave traces to indicate to what was done within a network and
+how. Removal of these files can occur during an intrusion, or as part of a post-intrusion process to minimize the
+adversary's footprint.
```
Comment (bm11100) [inline rules/windows/defense_evasion_process_termination_followed_by_deletion.toml:12]:
```suggestion
Identifies a process termination event quickly followed by the deletion of its executable file. Malware tools and other non-native
files dropped or created on a system by an adversary may leave traces to indicate to what occurred. Removal of these files can occur during an intrusion, or as part of a post-intrusion process to minimize the
adversary's footprint.
```

Comment (bm11100) [reply rules/windows/defense_evasion_process_termination_followed_by_deletion.toml:12]: re lint

File: rules/windows/defense_evasion_process_termination_followed_by_deletion.toml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+license = "Elastic License"
+name = "Process Termination Followed by Deletion"
+risk_score = 43
+rule_id = "09443c92-46b3-45a4-8f25-383b028b258d"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+type = "eql"
+
+query = '''
+sequence by host.id with maxspan=5s
```
Comment (bm11100) [inline rules/windows/defense_evasion_process_termination_followed_by_deletion.toml:26]: do you anticipate noise for this rule, or is the 5s timeframe enough to cut down FP's?

File: rules/windows/defense_evasion_process_termination_followed_by_deletion.toml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+Identifies process termination followed by the deletion of its executable file. Malware, tools, or other non-native
+files dropped or created on a system by an adversary may leave traces to indicate to what was done within a network and
+how. Removal of these files can occur during an intrusion, or as part of a post-intrusion process to minimize the
+adversary's footprint.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Process Termination Followed by Deletion"
```
Comment (bm11100) [inline rules/windows/defense_evasion_process_termination_followed_by_deletion.toml:18]:
```suggestion
name = "Process Termination followed by Deletion"
```

File: rules/windows/defense_evasion_process_termination_followed_by_deletion.toml (hunk 2 above)
Comment (Samirbous) [reply rules/windows/defense_evasion_process_termination_followed_by_deletion.toml:25]: 5s should reduce noise as well as excluding signed stuff

File: rules/windows/defense_evasion_process_termination_followed_by_deletion.toml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+Identifies a process termination event quickly followed by the deletion of its executable file. Malware tools and other non-native
+files dropped or created on a system by an adversary may leave traces to indicate to what occurred. Removal of these files can occur during an intrusion, or as part of a post-intrusion process to minimize the
+adversary's footprint.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Process Termination followed by Deletion"
+risk_score = 43
```
Comment (threat-punter) [inline rules/windows/defense_evasion_process_termination_followed_by_deletion.toml:18]:
```suggestion
risk_score = 47
```

Our other medium severity rules have a risk score of 47 if you want to align this one with those.

File: rules/windows/defense_evasion_process_termination_followed_by_deletion.toml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+type = "eql"
+
+query = '''
+sequence by host.id with maxspan=5s
+   [process where event.type == "end" and 
+    process.code_signature.trusted == false and
+    not process.executable : ("C:\\Windows\\SoftwareDistribution\\*.exe", "C:\\Windows\\WinSxS\\*.exe")
+   ] by  process.executable
```
Comment (threat-punter) [inline rules/windows/defense_evasion_process_termination_followed_by_deletion.toml:29]:
```suggestion
   ] by process.executable
```

**Review by threat-punter — APPROVED** [review]
Looks good. I made a couple of minor suggestions and the rule will need to be linted before merging 👍

File: rules/windows/defense_evasion_process_termination_followed_by_deletion.toml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+type = "eql"
+
+query = '''
+sequence by host.id with maxspan=5s
+   [process where event.type == "end" and 
+    process.code_signature.trusted == false and
+    not process.executable : ("C:\\Windows\\SoftwareDistribution\\*.exe", "C:\\Windows\\WinSxS\\*.exe")
+   ] by process.executable
+   [file where event.type == "deletion" and file.extension : "exe"] by file.path
```
Comment (dstepanic17) [inline rules/windows/defense_evasion_process_termination_followed_by_deletion.toml:30]: I noticed there were some other extensions (SCR, COM) in the issue, were those meant to be excluded?

Comment (Samirbous) [reply rules/windows/defense_evasion_process_termination_followed_by_deletion.toml:30]: forgot them, nice catch.

File: rules/windows/defense_evasion_process_termination_followed_by_deletion.toml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies a process termination event quickly followed by the deletion of its executable file. Malware tools and other non-native
+files dropped or created on a system by an adversary may leave traces to indicate to what occurred. Removal of these files can occur during an intrusion, or as part of a post-intrusion process to minimize the
+adversary's footprint.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
```
Comment (brokensound77) [inline rules/windows/defense_evasion_process_termination_followed_by_deletion.toml:14]: Will need to drop winlogbeat due to `process.code_signature.trusted`

**Review by brokensound77 — APPROVED** [review]
Just the one comment 👍

**Pull request merged**